ftw/upgrade/command/__init__.py

```diff
@@ -142,7 +144,25 @@ def __call__(self):
         args = self.parser.parse_args()
         configure_logging(args)
         setattr(args, 'parser', self.parser)
-        args.func(args)
+        if getattr(args, 'all_sites', False):
+            info = {}
+            while True:
+                try:
+                    with capture() as out:
+                        args.func(args)
+                    output = out.getvalue().strip()
+                    if getattr(args, 'json', False):
+                        # "[]" must be turned into [].
+                        output = json.loads(output)
+                    info[args.picked_site] = output
+                except StopIteration:
+                    break
+            if getattr(args, 'json', False):
+                print(json.dumps(info))
+            else:
+                print(info)
```

Comment:
BTW, I have made changes since the original pull request and kept rebasing and squashing it into one commit. That is nice for a final merge. But while you are reviewing this, extra commits could be nicer, so you can easily see the two or three lines that have changed, instead of possibly needing to look at all the changes again. So if you prefer that approach in the future, I'd be happy to oblige.